localstack/aws/api/firehose/__init__.py

```diff
@@ -0,0 +1,1860 @@
+import sys
+from datetime import datetime
+from typing import Dict, List, Optional
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+from localstack.aws.api import RequestContext, ServiceException, ServiceRequest, handler
+
+AWSKMSKeyARN = str
+AmazonopensearchserviceBufferingIntervalInSeconds = int
+AmazonopensearchserviceBufferingSizeInMBs = int
+AmazonopensearchserviceClusterEndpoint = str
+AmazonopensearchserviceDomainARN = str
+AmazonopensearchserviceIndexName = str
+AmazonopensearchserviceRetryDurationInSeconds = int
+AmazonopensearchserviceTypeName = str
+BlockSizeBytes = int
+BooleanObject = bool
+BucketARN = str
+ClusterJDBCURL = str
+CopyOptions = str
+DataTableColumns = str
+DataTableName = str
+DeliveryStreamARN = str
+DeliveryStreamName = str
+DeliveryStreamVersionId = str
+DescribeDeliveryStreamInputLimit = int
+DestinationId = str
+ElasticsearchBufferingIntervalInSeconds = int
+ElasticsearchBufferingSizeInMBs = int
+ElasticsearchClusterEndpoint = str
+ElasticsearchDomainARN = str
+ElasticsearchIndexName = str
+ElasticsearchRetryDurationInSeconds = int
+ElasticsearchTypeName = str
+ErrorCode = str
+ErrorMessage = str
+ErrorOutputPrefix = str
+HECAcknowledgmentTimeoutInSeconds = int
+HECEndpoint = str
+HECToken = str
+HttpEndpointAccessKey = str
+HttpEndpointAttributeName = str
+HttpEndpointAttributeValue = str
+HttpEndpointBufferingIntervalInSeconds = int
+HttpEndpointBufferingSizeInMBs = int
+HttpEndpointName = str
+HttpEndpointRetryDurationInSeconds = int
+HttpEndpointUrl = str
+IntervalInSeconds = int
+KinesisStreamARN = str
+ListDeliveryStreamsInputLimit = int
+ListTagsForDeliveryStreamInputLimit = int
+LogGroupName = str
+LogStreamName = str
+NonEmptyString = str
+NonEmptyStringWithoutWhitespace = str
+NonNegativeIntegerObject = int
+OrcRowIndexStride = int
+OrcStripeSizeBytes = int
+ParquetPageSizeBytes = int
+Password = str
+Prefix = str
+ProcessorParameterValue = str
+Proportion = float
+PutResponseRecordId = str
+RedshiftRetryDurationInSeconds = int
+RetryDurationInSeconds = int
+RoleARN = str
+SizeInMBs = int
+SplunkRetryDurationInSeconds = int
+TagKey = str
+TagValue = str
+Username = str
+
+
```

Comment:
Did you write out all the class definitions, or were they harvested from somewhere else? If it's the latter, could this be accomplished via some import?

Reply:
The content of this file is generated with our scaffold (`localstack/aws/scaffold.py`).
So it is generated using our own code generation when we start implementing a new service.
However, the implementation is tightly coupled to the generated code (and it takes some time to parse the specs and generate the code). Therefore we decided to commit the generated code in the repo.
In the future we will have a "generated code" warning on top of each file and might introduce a bot which automatically tries to regenerate the code (to automatically update to non-breaking changes in the specs) and creates PRs.

Reply:
cool!